source/gui/settingsDialogs.py

```diff
@@ -2935,213 +2934,6 @@ def onEnableControlsCheckBox(self, evt):
-
-		# Translators: This is a label for an edit field in add dictionary entry dialog.
-		patternLabelText = _("&Pattern")
-		self.patternTextCtrl=sHelper.addLabeledControl(patternLabelText, wx.TextCtrl)
-
-		# Translators: This is a label for an edit field in add dictionary entry dialog and in punctuation/symbol pronunciation dialog.
-		replacementLabelText = _("&Replacement")
-		self.replacementTextCtrl=sHelper.addLabeledControl(replacementLabelText, wx.TextCtrl)
-
-		# Translators: This is a label for an edit field in add dictionary entry dialog.
-		commentLabelText = _("&Comment")
-		self.commentTextCtrl=sHelper.addLabeledControl(commentLabelText, wx.TextCtrl)
-
-		# Translators: This is a label for a checkbox in add dictionary entry dialog.
-		caseSensitiveText = _("Case &sensitive")
-		self.caseSensitiveCheckBox=sHelper.addItem(wx.CheckBox(self,label=caseSensitiveText))
-
-		# Translators: This is a label for a set of radio buttons in add dictionary entry dialog.
-		typeText = _("&Type")
-		typeChoices = [DictionaryEntryDialog.TYPE_LABELS[i] for i in DictionaryEntryDialog.TYPE_LABELS_ORDERING]
-		self.typeRadioBox=sHelper.addItem(wx.RadioBox(self,label=typeText, choices=typeChoices))
-
-		sHelper.addDialogDismissButtons(wx.OK | wx.CANCEL, separated=True)
-
-		mainSizer.Add(sHelper.sizer, border=guiHelper.BORDER_FOR_DIALOGS, flag=wx.ALL)
-		mainSizer.Fit(self)
-		self.SetSizer(mainSizer)
-		self.setType(speechDictHandler.ENTRY_TYPE_ANYWHERE)
-		self.patternTextCtrl.SetFocus()
-		self.Bind(wx.EVT_BUTTON,self.onOk,id=wx.ID_OK)
-
-	def getType(self):
-		typeRadioValue = self.typeRadioBox.GetSelection()
-		if typeRadioValue == wx.NOT_FOUND:
-			return speechDictHandler.ENTRY_TYPE_ANYWHERE
-		return DictionaryEntryDialog.TYPE_LABELS_ORDERING[typeRadioValue]
-
-	def onOk(self,evt):
-		if not self.patternTextCtrl.GetValue():
-			# Translators: This is an error message to let the user know that the pattern field in the dictionary entry is not valid.
-			gui.messageBox(_("A pattern is required."), _("Dictionary Entry Error"), wx.OK|wx.ICON_WARNING, self)
-			self.patternTextCtrl.SetFocus()
-			return
-		try:
-			dictEntry = self.dictEntry = speechDictHandler.SpeechDictEntry(
-				self.patternTextCtrl.GetValue(),
-				self.replacementTextCtrl.GetValue(),
-				self.commentTextCtrl.GetValue(),
-				bool(self.caseSensitiveCheckBox.GetValue()),
-				self.getType()
-			)
-			dictEntry.sub("test")  # Ensure there are no grouping error (#11407)
-		except Exception as e:
-			log.debugWarning("Could not add dictionary entry due to (regex error) : %s" % e)
-			# Translators: This is an error message to let the user know that the dictionary entry is not valid.
-			gui.messageBox(_("Regular Expression error: \"%s\".")%e, _("Dictionary Entry Error"), wx.OK|wx.ICON_WARNING, self)
-			return
-		evt.Skip()
-
-	def setType(self, type):
-		self.typeRadioBox.SetSelection(DictionaryEntryDialog.TYPE_LABELS_ORDERING.index(type))
-
-
-class DictionaryDialog(SettingsDialog):

```

Comment:
While I agree `DictionaryDialog` should be eventually moved out of `gui\settingsDialogs` it would break backwards compatibility - various add-ons expect it either in the `gui` as it is star imported there or in the `gui\settingsDialogs`. Unles you  expect this to be merged before 2021.1 is finalized this part should be reverted IMO.

Reply:
On the other hand, 2021.1 already brings API changes.
Still, I restored as of 579cbf61 module-level import in `gui` of `DictionaryDialog`, as late-importing did not bring much benefit towards the wider API change.
@feerrenrut, may we please have your view here?

Reply:
I don't believe this will be in 2021.1, unless another dev sees reason to target this to beta instead. I think the changes in 579cbf6 are good, and should maintain the backwards compatibility. A comment should be added though to note this is a deprecation and should only be removed with a .1 release.

Reply:
we can import both of these removed classes into this module for now to maintain compatibility until 2022.1. Linting warnings can be ignored with a noqa comment to remove this in 2022.1

Reply:
Done as of 143bcde6a and fec7595b4.